pkg/libvmi/cloudinit/cloudinit.go

```diff
@@ -41,6 +41,12 @@ func WithNoCloudEncodedUserData(data string) NoCloudOption {
 	}
 }
 
+func WithEchoHelloNoCloudEncodedUserData() NoCloudOption {
```

Comment:
Thanks! It's great to see these awkward `echo hello` gone.
However, I'm not sure this function belongs in the general purpose pkg/libvmi/cloudinit/cloudinit.go; it seems specific to our tests and a peculiar requirement of Cirros that boots slowly if its cloudinit is empty.

Hence I'd put it in libvmifact (next to Cirros), call it `WithNonEmptyUserData` and explain when it should be used.

Reply:
Moved the function to `tests/libvmifact/factory.go` , about the naming I would like to keep the EchoHello inside, so people can understand from the function naming what goes into the cloud-init section, so changed to `WithEchoHelloNoUserData` for it to be shorter